Source/JavaScriptCore/runtime/JSBigInt.h

```diff
@@ -461,6 +461,9 @@ class JSBigInt final : public JSCell {
         return static_cast<int64_t>(toBigUInt64Heap(bigInt.asHeapBigInt()));
     }
 
+    // Use int64 maximum as the flag to indicate to bigint to int64 conversion failure. Note that it's only used for converting JSBigInt to int64_t in DFG/FTL tier for BigInt64.
+    static constexpr const int64_t notInt64 = std::numeric_limits<int64_t>::max();
```

Comment:
Let's just do inlined check for conversion. And for all Int64 operations, you can check overflow flag in JIT and OSR exit. So we should not have this.